### PR TITLE
feat: Generic data processor

### LINKED ISSUE #113: [FEAT] auxiliary processing functions
Support synthesizers with a `DataProcessor` class for synth training and inverse transforms for the synthetic samples.
Synthesizers sample method should leverage the inverse_transform of the `DataProcessor` on the synthetic data to return it in the original data format.

### PR DESCRIPTION
**Pull request by jfsantos-ds**
Adds a generic data processor to cover synthesizer examples.

This processor is integrated in the ydata synthetic examples to simplify the data processing pipelines.
The preprocessor can be passed to the synthesizers in order to enable the following behaviors:
- Transform the raw data directly in synthesizer train method
- Make the inverse_transform method of the preprocessor available to the synthesizer sample method, so that synthetic observations can be returned directly in the original format

Closes #113

### CODE REVIEW

File: src/ydata_synthetic/preprocessing/regular/data_processor.py [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+    """
+    Main class for Data Preprocessing. It is a base version.
+    It works like any other transformer in scikit lear with the methods fit, transform and inverse transform.
+    Args:
+        num_cols (list of strings):
+            List of names of numerical columns
+        cat_cols (list of strings):
+            List of names of categorical columns
+    """
+    @typechecked
```
Comment (fabclmnt) [inline src/ydata_synthetic/preprocessing/regular/data_processor.py:21]: You can perform the _typechecked_ decorator at the level of the class and only use the _typeguard_ignore_ decorator in the methods that you don't want to get checked based on the annotations.

Comment (jfsantos-ds) [reply src/ydata_synthetic/preprocessing/regular/data_processor.py:21]: Thanks, nice tip, makes it more readable

File: src/ydata_synthetic/preprocessing/regular/data_processor.py [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+    """
+    Main class for Data Preprocessing. It is a base version.
+    It works like any other transformer in scikit lear with the methods fit, transform and inverse transform.
+    Args:
+        num_cols (list of strings):
+            List of names of numerical columns
+        cat_cols (list of strings):
+            List of names of categorical columns
+    """
+    def __init__(self, *, num_cols: List[str] = None, cat_cols: List[str] = None):
```
Comment (fabclmnt) [inline src/ydata_synthetic/preprocessing/regular/data_processor.py:21]: Missing docstrings

**Review by fabclmnt — APPROVED** [review]
🚀

**Pull request merged**